CMakeLists.txt

```diff
@@ -1018,21 +1018,22 @@ configure_file(
 if(CONFIG_CPU_HAS_MPU AND CONFIG_USERSPACE)
 
     if(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT AND CONFIG_APP_SHARED_MEM)
-        set(GEN_APP_SMEM $ENV{ZEPHYR_BASE}/scripts/gen_app_smem.py)
+        set(GEN_APP_SMEM $ENV{ZEPHYR_BASE}/scripts/gen_app_partitions.py)
```

Comment:
${ZEPHYR_BASE} must be used instead of $ENV{ZEPHYR_BASE} for portability reasons (I know this bug is not introduced by this PR).